dspace-api/src/main/java/org/dspace/authenticate/SamlAuthentication.java

```diff
@@ -0,0 +1,750 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.authenticate;
+
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.authenticate.factory.AuthenticateServiceFactory;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.MetadataField;
+import org.dspace.content.MetadataSchema;
+import org.dspace.content.MetadataSchemaEnum;
+import org.dspace.content.NonUniqueMetadataException;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.content.service.MetadataSchemaService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.Group;
+import org.dspace.eperson.factory.EPersonServiceFactory;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.eperson.service.GroupService;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+/**
+ * SAML authentication for DSpace.
+ *
+ * @author Ray Lee
+ */
+public class SamlAuthentication implements AuthenticationMethod {
+    private static final Logger log = LogManager.getLogger(SamlAuthentication.class);
+
+    // Additional metadata mappings.
+    protected Map<String, String> metadataHeaderMap = null;
+
+    // Maximum length for ePerson fields.
+    protected final int NAME_MAX_SIZE = 64;
```

Comment:
`EPerson` name (full = first + last, or email if null) is stored in metadata and so technically has a max size of `Length.LONG32` - the email address limit is 64 characters, though email length is not validated later on in the register method.
Is this check maybe because there are situations where the first and last names are null, the `getFullName()` method then defaults to email, so you need to validate against email max length? Or can we increase the maximum name length?

Reply:
I admit that I just cribbed this from ShibAuthentication, which I used as a model for SamlAuthentication. I wonder if there was a restriction in place when ShibAuthentication was implemented?